PracticeProject/API/UserAPI.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  UserAPI.swift
+//  PracticeProject
+//
+//  Created by Gnanavel on 31/03/22.
+//
+
+import Foundation
+
+class UserAPI : API {
+    
+    private let apiBaseURL = "https://jsonplaceholder.typicode.com"
+    
+    func fetchUsers(excludingUserWithID: String?, success: @escaping (UsersList?) -> Void, failure: @escaping (FetchError?) -> Void) {
+        guard Reachability.isConnectedToNetwork() else {
+            failure("Check your network connection, please try again.")
+            return
+        }
+        guard let url = URL(string: "\(apiBaseURL)/users") else { return }
+        URLSession.shared.dataTask(with: url) { data, response, error in
+            guard let data = data else {
```

Comment:
What happens if this `dataTask` callback is invoked after `self` (UserAPI instance) is deallocated? 
Compare the lifecycle of this object in comparison to this `dataTask` callback block.

Reply:
@nets-dev , Yes, It's gets deallocated before the completion handler can be called. 
The completion handler can't retain our dataTask callback and after execution of run loop, the garbage collector will deallocate. Kindly let me know if I did misunderstand.

Reply:
Swift uses ARC, not garbage collector. Can you answer the question with focus on "what happens" in this scenario? 
(Think about the lifecycle of all objects and closures involved in this scenario)